mitosheet/src/mito/utils/keyboardShortcuts.tsx

```diff
@@ -0,0 +1,88 @@
+import { ActionEnum, KeyboardShorcut } from "../types"
+import { Actions } from "./actions"
+
+/**
+ * A list of all keyboard shortcuts in the app.
+ */
+export const keyboardShortcuts: KeyboardShorcut[] = [
+    {
+        macKeyCombo: {
+            metaKey: true,
+            key: ['c']
+        },
+        winKeyCombo: {
+            ctrlKey: true,
+            key: ['c']
+        },
+        action: ActionEnum.Copy
+    },
+    {
+        macKeyCombo: {
+            metaKey: true,
+            key: ['f']
+        },
+        winKeyCombo: {
+            ctrlKey: true,
+            key: ['f']
+        },
+        preventDefaultAndStopPropagation: true,
+        jupyterLabAction: 'mitosheet:open-search',
+        action: ActionEnum.OpenSearch
+    },
+    {
+        macKeyCombo: {
+            metaKey: true,
+            key: ['z']
+        },
+        winKeyCombo: {
+            ctrlKey: true,
+            key: ['z']
+        },
+        jupyterLabAction: 'mitosheet:mito-undo',
+        action: ActionEnum.Undo
+    },
+    {
+        macKeyCombo: {
+            metaKey: true,
+            key: ['y']
+        },
+        winKeyCombo: {
+            ctrlKey: true,
+            key: ['y']
+        },
+        preventDefaultAndStopPropagation: true,
+        action: ActionEnum.Redo
+    }
+]
+
+/**
+ * Handles keyboard shortcuts. If a keyboard shortcut is pressed, the corresponding action is executed.
+ * @param e The keyboard event.
+ * @param actions The actions object.
+ */
+export const handleKeyboardShortcuts = (e: React.KeyboardEvent, actions: Actions) => {
+    const operatingSystem = window.navigator.userAgent.toUpperCase().includes('MAC')
+                    ? 'mac'
+                    : 'windows'
+
+    const shortcut = keyboardShortcuts.find(shortcut => {
+        const keyCombo = operatingSystem === 'mac' ? shortcut.macKeyCombo : shortcut.winKeyCombo
+        // If the special key combination doesn't match, return false.
+        if (!!e.metaKey !== !!keyCombo.metaKey ||
```

Comment:
Ohh yeah it's just in case one of them is undefined and the other is false. Doing that converts undefined to false. 
```
!undefined = true
!!undefined = false
```